### PR TITLE
ii-6413 Ensure the ability to fuse metrics in the Kafka agent

### PR DESCRIPTION
**Pull request by zhangzinan**

### CODE REVIEW

File: kafka2/README.md [hunk 1]
```diff
@@ -64,7 +64,7 @@ python getmessages_kafka2.py -t
 
 ###### If satisfied with the output, configure the agent to run continuously:
 ```bash
-sudo ./setup/cron-config.sh
```
Comment (ptmr3) [inline kafka2/README.md:67]: @zhangzinan Why this change?

File: kafka2/config.ini.template [hunk 2]
```diff
@@ -73,6 +73,12 @@ device_field =
 # multiple fields are separated by commas. a field can be named with the syntax `<name>:<value>` or `<name>:=<value>`, where `<name>` and `<value>` can each be either a literal value (`name:value`) or formatted (`total time [{step}]:={timing.end}-{timing.start}`). Use `:=` as a separator to treat `<value>` as a mathematical formula, which must be parseable by `numexpr.evaluate()`.
 data_fields =
 
+## metric buffer
+# if set this field, agent will send data at once when all metrics of instance is collected. example: metric1,metric2,...
+all_metrics =
+# required. must greater than 0
```
Comment (yqian10) [inline kafka2/config.ini.template:79]: please put more explanation to metric_buffer_size_mb

File: kafka2/getmessages_kafka2.py [hunk 3]
```diff
@@ -1331,16 +1377,58 @@ def send_metric(timestamp, field_name, data, instance, device=''):
...
+            # move last metric dict from buffer_dict to current_dict
+            if get_json_size_bytes(metric_buffer['buffer_dict']) >= agent_config_vars['metric_buffer_size'] or (
+                    time.time() - metric_buffer['start_time']) >= agent_config_vars['metric_buffer_keep_last_time']:
+                # if the earliest metric data is exceed the butter start_time, the move those data to track
+                while metric_buffer['buffer_ts_list'] \
+                        and metric_buffer['buffer_ts_list'][-1] < metric_buffer['start_time'] * 1000:
+                    ts_str = str(metric_buffer['buffer_ts_list'].pop())
+                    track['current_dict'][ts_str] = metric_buffer['buffer_dict'].pop(ts_str)
+                # update the buffer start_time
+                metric_buffer['start_time'] += if_config_vars['sampling_interval']
```
[the remark this answers is not in the document]
Comment (zhangzinan) [reply kafka2/getmessages_kafka2.py]: new commit remove this.

File: kafka2/README.md (hunk 1 above)
Comment (zhangzinan) [reply kafka2/README.md:67]: 1. This agent will exit if no msg received after a certain time. So need monit to start it again.
2. This agent need only start one process to consume one topic

File: kafka2/getmessages_kafka2.py [hunk 4]
```diff
@@ -1668,7 +1757,15 @@ def initialize_api_post_data():
     print_summary_info()
 
     # start data processing
+    process_list = []
     for i in range(0, cli_config_vars['threads']):
-        Process(target=initialize_data_gathering,
-                args=(i,)
-                ).start()
+        p = Process(target=initialize_data_gathering,
```
[the remark this answers is not in the document]
Comment (zhangzinan) [reply kafka2/getmessages_kafka2.py:1738]: Please refer to the official documents

File: kafka2/README.md (hunk 1 above)
Comment (ptmr3) [reply kafka2/README.md:67]: We need to check the cron config and make sure it does the following:
- Check every 60 seconds to see if the process is running
-- If the process is not running, start it.
-- else, do nothing

Comment (yqian10) [reply kafka2/README.md]: when I install the cron, and it seems does not doing what we want: 
Screenshot from 2020-04-22 11-23-52
